addons/mail/static/tests/mock_server/mail_mock_server.js

```diff
@@ -66,52 +186,72 @@ async function channelAttachments(request) {
 /** @type {RouteCallback} */
-async function channelJoinCall(request) {
+async function channel_call_join(request) {
+    /** @type {import("mock_models").DiscussChannel} */
+    const DiscussChannel = this.env["discuss.channel"];
+    /** @type {import("mock_models").DiscussChannelMember} */
+    const DiscussChannelMember = this.env["discuss.channel.member"];
+    /** @type {import("mock_models").DiscussChannelRtcSession} */
+    const DiscussChannelRtcSession = this.env["discuss.channel.rtc.session"];
+
     const { channel_id } = await parseRequestParams(request);
-    const memberOfCurrentUser =
-        this.env["discuss.channel.member"]._getAsSudoFromContext(channel_id);
-    const sessionId = this.env["discuss.channel.rtc.session"].create({
+    const memberOfCurrentUser = DiscussChannel._find_or_create_member_for_self(channel_id);
+    const sessionId = DiscussChannelRtcSession.create({
         channel_member_id: memberOfCurrentUser.id,
         channel_id, // on the server, this is a related field from channel_member_id and not explicitly set
         guest_id: memberOfCurrentUser.guest_id[0],
         partner_id: memberOfCurrentUser.partner_id[0],
     });
-    const channelMembers = this.env["discuss.channel.member"]._filter([
-        ["channel_id", "=", channel_id],
-    ]);
-    const rtcSessions = this.env["discuss.channel.rtc.session"]._filter([
+    const channelMembers = DiscussChannelMember._filter([["channel_id", "=", channel_id]]);
+    const rtcSessions = DiscussChannelRtcSession._filter([
         ["channel_member_id", "in", channelMembers.map((channelMember) => channelMember.id)],
     ]);
     return {
         iceServers: false,
         rtcSessions: [
-            ["ADD", rtcSessions.map((rtcSession) => this._mailRtcSessionFormat(rtcSession.id))],
+            [
+                "ADD",
+                rtcSessions.map((rtcSession) =>
+                    DiscussChannelRtcSession._mail_rtc_session_format(rtcSession.id)
+                ),
+            ],
         ],
         sessionId: sessionId,
     };
 }
 
+registerRoute("/mail/rtc/channel/leave_call", channel_call_leave);
 /** @type {RouteCallback} */
-async function channelLeaveCall(request) {
+async function channel_call_leave(request) {
+    /** @type {import("mock_models").BusBus} */
+    const BusBus = this.env["bus.bus"];
+    /** @type {import("mock_models").DiscussChannel} */
+    const DiscussChannel = this.env["discuss.channel"];
+    /** @type {import("mock_models").DiscussChannelMember} */
+    const DiscussChannelMember = this.env["discuss.channel.member"];
+    /** @type {import("mock_models").DiscussChannelRtcSession} */
+    const DiscussChannelRtcSession = this.env["discuss.channel.rtc.session"];
+    /** @type {import("mock_models").MailGuest} */
+    const MailGuest = this.env["mail.guest"];
+    /** @type {import("mock_models").ResPartner} */
+    const ResPartner = this.env["res.partner"];
```

Comment:
Readability sucks a bit in the header of function code, but the bottom is simpler and essentially type-safe.
There's more gain than loss with them, so that's why I did that.

A much less verbose solution would be to augment `MockServerEnvironment` to take `mock_models.d.ts` into account. I tried a bit to do that but I couldn't make it work, so plan B was that verbose approach.